demo-linux/demo.cpp

```diff
@@ -120,19 +144,71 @@ void demo()
             exit(-1);
         }
     }
+
+    writeToEEPROM();
 }
 
 void usage()
 {
-    printf("Usage: $ demo-linux [--count n] [--log-level DEBUG|INFO|ERROR|NEVER]\n");
+    printf("Usage: $ demo-linux [--count n] [--log-level DEBUG|INFO|ERROR|NEVER] [--integration-time-ms n] [--raman-mode]\n");
     exit(1);
 }
 
 void parseArgs(int argc, char** argv)
 {
     for (int i = 1; i < argc; i++)
     {
-        if (!strcmp(argv[i], "--count"))
+        printf("current arg is %s\n", argv[i]);
+        if (!strcmp(argv[i], "--integration-time-ms")) 
+        {
+            if (i + 1 < argc)
+            {
+                int ms = atoi(argv[++i]);
+                wp_set_integration_time_ms(specIndex,ms);
+            }
+            else
+            {
+                printf("not enough args according to integration time\n");
+                usage();
+            }
+        }
+        else if (!strcmp(argv[i], "--raman-mode"))
+        {
```

Comment:
Merging in a comment I'd left at the file level if that isn't automatically sent as an email notification.

I have several issues here.

1. The purpose of this function is to parse command-line arguments.  Not to execute a series of lengthy spectroscopic measurements.  This function should parse the command-line arguments into accessible variables and that is it.

2. By stuffing all this test logic into the parseArgs function, you are imposing complicated requirements on the user to enter the arguments in a specific sequence.  For instance, if they type "--raman-mode --integration-time-ms 100 --count 5 --log-level debug" they will get extremely different behavior than if they typed "--log-level debug --count 5 --integration-time-ms 100 --raman-mode".  That is part of why this function should only do one thing.

3. You're now calling init() multiple times.  That is unnecessary and probably not a good thing.